src/segmantic/seg/evaluation.py

```diff
@@ -0,0 +1,242 @@
+
+
+def hausdorff_surface_distance(y_pred: itkImage, y_ref: itkImage) -> Dict[str, float]:
+    """Compute symmetric surface distances between two binary masks
+
+    Args:
+        y_pred (itkImage): predicted segmentation
+        y_ref (itkImage): reference segmentation
+
+    Returns:
+        Dict[str, float]: keys are 'mean', 'median', 'std', 'max'
+    """
+    seg_surface = itk.binary_contour_image_filter(y_pred, foreground_value=1)
+    ref_surface = itk.binary_contour_image_filter(y_ref, foreground_value=1)
+
+    seg_distance = itk.signed_maurer_distance_map_image_filter(
+        y_pred, use_image_spacing=True, squared_distance=False, inside_is_positive=False
+    )
+    ref_distance = itk.signed_maurer_distance_map_image_filter(
+        y_ref, use_image_spacing=True, squared_distance=False, inside_is_positive=False
+    )
+
+    # get distance at contour of foreground label
+    ref2seg_distance = np.multiply(ref_surface, seg_distance)
+    ref2seg_distance = ref2seg_distance[np.nonzero(ref_surface)]
+
+    seg2ref_distance = np.multiply(seg_surface, ref_distance)
+    seg2ref_distance = seg2ref_distance[np.nonzero(seg_surface)]
+
+    # compute statistics on symmetric distances (both directions)
+    all_surface_distances = np.concatenate(
+        (ref2seg_distance, seg2ref_distance), axis=None
+    )
+    all_surface_distances = np.abs(all_surface_distances)
+
+    mean_surface_distance = np.mean(all_surface_distances)
+    median_surface_distance = np.median(all_surface_distances)
+    std_surface_distance = np.std(all_surface_distances)
+    max_surface_distance = np.max(all_surface_distances)
+    return {
+        "mean": mean_surface_distance,
+        "median": median_surface_distance,
+        "std": std_surface_distance,
+        "max": max_surface_distance,
+    }
+
+
+def hausdorff_pointwise_distance(y_pred: itkImage, y_ref: itkImage) -> Dict[str, float]:
+    """Compute symmetric point-wise distances between two binary masks
+
+    Args:
+        y_pred (itkImage): predicted segmentation
+        y_ref (itkImage): reference segmentation
+
+    Returns:
+        Dict[str, float]: keys are 'mean', 'median', 'std', 'max'
+    """
+    seg_distance = itk.signed_maurer_distance_map_image_filter(
+        y_pred, use_image_spacing=True, squared_distance=False, inside_is_positive=False
+    )
+    ref_distance = itk.signed_maurer_distance_map_image_filter(
+        y_ref, use_image_spacing=True, squared_distance=False, inside_is_positive=False
+    )
+
+    # get distance inside foreground label
+    ref2seg_distance = seg_distance[np.nonzero(y_ref)]
+    seg2ref_distance = ref_distance[np.nonzero(y_pred)]
+
+    # compute statistics on symmetric distances (both directions)
+    all_surface_distances = np.concatenate(
+        (ref2seg_distance, seg2ref_distance), axis=None
+    )
+    all_surface_distances[all_surface_distances <= 0.0] = 0.0
+
+    mean_surface_distance = np.mean(all_surface_distances)
```

Comment:
i vaguely recall a method that would compute all this in one go instead of having to sweep several times over the array

Reply:
as a one liner, what about this?

```python
def stats( data: np.array, aggregators: Sequence[Callable] = None )
    if aggregators is None:
        aggregators = (np.mean, np.std, np.max, np.min, np.median)

return { f.__name__: f(data) for f in aggregators }

Reply:
maybe, but I don't find it more readable, and its not a one-liner if you count the function `stats` ;-)